src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java

```diff
@@ -486,6 +487,7 @@ private static DataGroup getFitsImageAsTable(BasicHDU<?> hdu,
                 }
 
                 DataGroup dataGroup = new DataGroup(desc, dataTypes);
+                dataGroup.setInitCapacity(data[0].length);
```

Comment:
What did you not add a `setInitCapacity` after new `new DataGroup` in this file?

Reply:
When not set, it defaults to 1000.